core/src/main/scala/cats/data/ZipStream.scala

```diff
@@ -0,0 +1,32 @@
+package cats.data
+
+import cats.{Alternative, CommutativeApplicative, Eq}
+import cats.instances.stream._
+
+class ZipStream[A](val value: Stream[A]) extends AnyVal
+
+object ZipStream {
+
+  def apply[A](value: Stream[A]): ZipStream[A] = new ZipStream(value)
+
+  implicit val catsDataAlternativeForZipStream: Alternative[ZipStream] with CommutativeApplicative[ZipStream] =
+    new Alternative[ZipStream] with CommutativeApplicative[ZipStream] {
```

Comment:
This should be lawful right? but we can't test it...that's awkward. how about moving it to alleycats?

Reply:
Awkward indeed, I'm not sure if alleycats is the right place, since they can be proven to be lawful in e.g. Haskell, but I agree that it's awkward.

Maybe we should move the `CommutativeApplicative` instance to Alleycats and keep the `CommutativeApply` instance here?

Reply:
I thought about that but it would cause implicit conflict if you import both right? Unless we do the export hook trick which is kind of deprecated. Maybe we just document a bit more.